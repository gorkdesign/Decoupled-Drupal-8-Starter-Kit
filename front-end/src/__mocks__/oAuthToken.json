{
  "data": {
    "expires_in": 300,
    "access_token":
      "asdf",
    "refresh_token":
      "asdfasdf"
  }
}
