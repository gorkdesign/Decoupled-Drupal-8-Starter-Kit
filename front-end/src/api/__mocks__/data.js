export const CURRENT_USER_QUERY_DATA = {
  data: {
    user: {
      uid: 1,
      uuid: '4e9af657-689b-4c06-8721-e267914f2255',
    },
  },
};

export const ARTICLES_BY_USER_DATA = {
  data: {
    user: {
      uid: 1,
      nodes: {
        articles: [
          {
            author: {
              name: 'admin',
            },
            title: 'hello article update',
            body: null,
            nid: 13,
            uuid: '79502776-61f8-4c48-b464-d94eebe0e01b',
            images: [],
          },
          {
            author: {
              name: 'admin',
            },
            title: 'asdfads',
            body: {
              value: '<p>asdfasdf</p>\r\n',
            },
            nid: 14,
            uuid: 'be510cd5-645e-4f72-8baa-06cf89f53f84',
            images: [],
          },
        ],
      },
    },
  },
};


export default {};
