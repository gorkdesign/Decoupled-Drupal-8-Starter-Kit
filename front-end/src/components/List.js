import React, { Component } from 'react';
import PropTypes, { arrayOf, shape } from 'prop-types';
import { Button, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';
import { TransitionGroup, CSSTransition } from 'react-transition-group';
import MdAdd from 'react-icons/lib/md/add';

import ARTICLE_SHAPE from '../utils/articlePropType';
import Card from './Card';

const Fade = ({ children, ...props }) => (
  <CSSTransition
    {...props}
    timeout={1000}
    classNames="fade"
  >
    {children}
  </CSSTransition>
);
Fade.propTypes = {
  children: PropTypes.node.isRequired,
};

const ListItems = ({ articles, selectHandler, deleteHandler }) => articles.map(page => (
  <Fade duration={1000} key={page.nid} timeout={{ enter: 0, exit: 1000 }}>
    <Card
      page={page}
      selectHandler={selectHandler}
      deleteHandler={(event) => { deleteHandler(event, page.nid); }}
    />
  </Fade>
));

class List extends Component {
  static getDerivedStateFromProps(nextProps, prevState) {
    if (nextProps.articles.length !== prevState.articles.length) {
      return {
        articles: nextProps.articles,
      };
    }
    return null;
  }

  state = {
    articles: this.props.articles,
  }

  handleSearchInput = (target) => {
    const searchTerm = target.value;

    const { articles } = this.props;
    const searchedArticles = articles.map((page) => {
      const title = page.title.toLowerCase();
      const searchTermLower = searchTerm.toLowerCase();
      return title.includes(searchTermLower) ? page : null;
    });
    const filteredArticles = searchedArticles.filter(n => n != null);

    if (filteredArticles.length > 0) {
      this.setState({
        articles: filteredArticles,
      });
    } else if (target.value === '') {
      this.setState({
        articles: this.props.articles,
      });
    } else {
      this.setState({
        articles: [],
      });
    }
  }

  render() {
    const {
      addHandler, isModalVisible, onDeleteModalToggle, onDeleteModalOk, selectHandler, deleteHandler,
    } = this.props;

    return (
      <div className="">
        <div className="container">
          <div
            role="button"
            tabIndex={0}
            onKeyUp={() => { addHandler(); }}
            className="py-3"
            onClick={() => { addHandler(); }}
          >
            <div className="add">
              <MdAdd />
              <h2 className="card-title">Add</h2>
            </div>
          </div>

          <div className="input-group input-group-lg">
            <div className="input-group-prepend">
              <span className="input-group-text" id="inputGroup-sizing-lg">Search By Title</span>
            </div>
            <input onChange={({ target }) => this.handleSearchInput(target)} placeholder="Content Title..." type="text" className="form-control search-bar" aria-label="Large" aria-describedby="inputGroup-sizing-sm" />
          </div>

          {
            this.state.articles.length > 0 ?
              <TransitionGroup className="item-list">
                <ListItems
                  articles={this.state.articles}
                  selectHandler={selectHandler}
                  deleteHandler={deleteHandler}
                />
              </TransitionGroup>
          : <div className="no-result-text h1 text-center">No Results Found</div>
          }

          <Modal isOpen={isModalVisible} toggle={onDeleteModalToggle} backdrop>
            <ModalHeader toggle={onDeleteModalToggle}>Confirmation</ModalHeader>
            <ModalBody>
              Are you sure you want to remove this?
            </ModalBody>
            <ModalFooter>
              <Button color="secondary" onClick={onDeleteModalToggle}>Cancel</Button>
              {' '}
              <Button color="primary" onClick={onDeleteModalOk}>Delete</Button>
            </ModalFooter>
          </Modal>
        </div>
      </div>
    );
  }
}

List.propTypes = {
  articles: arrayOf(shape(ARTICLE_SHAPE).isRequired).isRequired,
  isModalVisible: PropTypes.bool.isRequired,
  addHandler: PropTypes.func.isRequired,
  onDeleteModalToggle: PropTypes.func.isRequired,
  onDeleteModalOk: PropTypes.func.isRequired,
};

export default List;
