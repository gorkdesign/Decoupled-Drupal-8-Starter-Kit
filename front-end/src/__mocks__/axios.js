import * as oauthTokens from './oAuthToken.json';

const URL = process.env.REACT_APP_HOST_DOMAIN ? process.env.REACT_APP_HOST_DOMAIN : '';
const OAUTH_ENDPOINT = `${URL}/oauth/token`;
const CSRF_ENDPOINT = `${URL}/session/token`;

module.exports = {
  get: jest.fn((url) => {
    switch (url) {
      case CSRF_ENDPOINT: return Promise.resolve({ data: 'asdf' });
      default: {
        const error = new Error('url not reachable');
        return Promise.reject(error);
      }
    }
  }),
  post: jest.fn((url) => {
    switch (url) {
      case OAUTH_ENDPOINT: {
        const result = Promise.resolve(oauthTokens);
        return result;
      }
      default:
        return null;
    }
  }),
};
