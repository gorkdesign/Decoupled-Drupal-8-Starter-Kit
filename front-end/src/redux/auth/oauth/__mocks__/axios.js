import * as oauthTokens from './oAuthToken.json';

const URL = process.env.REACT_APP_HOST_DOMAIN ? process.env.REACT_APP_HOST_DOMAIN : '';
const OAUTH_ENDPOINT = `${URL}/oauth/token`;

module.exports = {
  post: jest.fn((url) => {
    switch (url) {
      case OAUTH_ENDPOINT:
        return Promise.resolve(oauthTokens);
      default:
        return null;
    }
  }),
};
