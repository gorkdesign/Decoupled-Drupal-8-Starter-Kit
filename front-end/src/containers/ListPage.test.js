import React from 'react';
import { mount, shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import createSagaMiddleware from 'redux-saga';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import List from './ListPage';
import * as data from '../api/__mocks__/data';


describe('ListPage', () => {
  const sagaMiddleware = createSagaMiddleware();
  const mockStore = configureMockStore([sagaMiddleware]);
  const store = mockStore({});

  const initialPropsState = {
    articles: data.ARTICLES_BY_USER_DATA.data.user.nodes.articles,
  };

  let props;
  let mountedLogin;
  const ListPage = (modifiedProps = initialPropsState) => {
    props = {
      ...initialPropsState,
      ...modifiedProps,
    };

    if (!mountedLogin) {
      mountedLogin = mount(<List {...props} />);
    }
    return mountedLogin;
  };

  beforeEach(() => {
    props = initialPropsState;
    mountedLogin = undefined;
  });

  describe('Snapshots', () => {
    const output = shallow(<Provider store={store}><ListPage {...initialPropsState} /></Provider>);
    it('should render correctly', () => {
      expect(shallowToJson(output)).toMatchSnapshot();
    });
  });


  // describe('ListPageArticle differing props', () => {
  //   it('Should NOT display the login form if authenticated - SHOULD REDIRECT', () => {
  //     const newProps = { ...initialPropsState };
  //     newProps.isAuthenticated = true;
  //     const wrapper = shallow(<Login {...newProps} />);
  //     expect(wrapper.find(Redirect).props().to).toBe('/');
  //   });
  // });
});
