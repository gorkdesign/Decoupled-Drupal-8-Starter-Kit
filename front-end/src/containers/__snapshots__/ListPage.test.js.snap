// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPage Snapshots should render correctly 1`] = `
<ListPage
  articles={
    Array [
      Object {
        "author": Object {
          "name": "admin",
        },
        "body": null,
        "images": Array [],
        "nid": 13,
        "title": "hello article update",
        "uuid": "79502776-61f8-4c48-b464-d94eebe0e01b",
      },
      Object {
        "author": Object {
          "name": "admin",
        },
        "body": Object {
          "value": "<p>asdfasdf</p>
    ",
        },
        "images": Array [],
        "nid": 14,
        "title": "asdfads",
        "uuid": "be510cd5-645e-4f72-8baa-06cf89f53f84",
      },
    ]
  }
/>
`;
